src/cljs/rems/table.cljs

```diff
@@ -63,18 +67,60 @@
   (reduce (fn [items {:keys [sort-column sort-order]}]
             (apply-sorting column-definitions sort-column sort-order items)) items initial-sort))
 
-(defn matches-filter [column-definitions col filter-value item]
-  (let [actual-value (str (column-filter-value column-definitions col item))]
-    (str/includes? (str/lower-case actual-value)
-                   (str/lower-case filter-value))))
-
-(defn matches-filters [column-definitions filters item]
-  (every? (fn [[col filter-value]] ()
-            (matches-filter column-definitions col filter-value item))
-          filters))
+(defn match-filter? [column-definitions filter-word item]
+  (some (fn [col]
+          (str/includes? (str/lower-case (column-filter-value column-definitions col item))
+                         (str/lower-case filter-word)))
+        (keys column-definitions)))
+
+(deftest test-match-filter
+ (is (true? (match-filter? {:id {:value :id}
+                            :description {:value :description}}
+                           "bar"
+                           {:id "foo"
+                            :description "bar"}))
+     "matches")
+ (is (not (match-filter? {:id {:value :id}
+                          :description {:value :its-a-trap
+                                        :filter-value :description}}
+                         "willnotmatch"
+                         {:id "foo"
+                          :its-a-trap "willnotmatch"
+                          :description "bar"}))
+     "doesn't match with value if filter-value is provided"))
+
+(defn match-filters? [column-definitions filters item]
+  (every? (fn [filter-word]
+            (match-filter? column-definitions filter-word item))
+          (str/split filters #"\s+")))
```

Comment:
I don't think this is a good place for the `str/split`. it should be done earlier so that filters can be a seq already.

Reply:
At most it could be moved one level up to apply-filters, which doesn't make much of a difference. Any higher and we are in representational logic that shouldn't really care how filter logic wants to interpret the filter commands.

Reply:
I think `str/split` belongs to the representational layer since it's parsing user input and not in the logic layer here. That's why I suggest that you move the parsing up to the border where the input callback is. Internally you can store the filters in a more optimal format for search and do less processing every call.

Reply:
I think it might make sense to refactor this later anyway since there is very little value in having `:filterable?` and `:sortable?` separately compared to only specifying `:filter-value` and `:sort-value` where necessary.

Reply:
`:filter-value` and `:sort-value` fall back to `:value`, so `:filterable?` and `:sortable?` exist to disable the default behavior.

Reply:
Yes but that is also suspect.